prost-build/src/code_generator.rs

```diff
@@ -461,9 +472,14 @@ impl<'a> CodeGenerator<'a> {
             }
         }
 
+        if is_well_known_ty {
```

Comment:
Why is it important to know whether it is a well known type? Or asked differently: Why is well known type special?

Reply:
I mentioned this quite shortly in `prost-types/serde.rs`:

```
// Due to the orphan rules we can't really implement prost::serde::CustomSerialize for our
// well-known types that are primitives. So we have to use a local wrapper type instead.
```

Tbqh, I don't remember the exact details, but I went with putting these `CustomSerialize/CustomDeserialize` impls for scalar types in the prost-types crate instead of the main `prost`crate due to separation concerns. And I generally though it would give us more flexibility to avoid the orphan rules whenever we'd have to provide custom serialization impls.

Reply:
But I do realized now that this approach might be flawed because now you'd need to always include the `prost-types` crate to have serialization working for all "built-in" types. That's certainly a no-go, so I'll try to put back the impls back into `prost` and will eventually remove the whole "is well known type" treatment.